src/components/pages/profile/file-input.tsx

```diff
@@ -0,0 +1,114 @@
+'use client'
+
+import ImageCard from '@/components/pages/auth/image-card'
+import { Icons } from '@/components/ui/icons'
+import { Progress } from '@/components/ui/progress'
+import { cn } from '@/lib/utils'
+import { FC, FormEvent, useRef, useState } from 'react'
+
+type FileInputProps = {
+  className?: string
+}
+
+const FileInput: FC<FileInputProps> = ({ className }) => {
+  const fileInputRef = useRef<HTMLInputElement>(null)
+
+  const [_, setImage] = useState<File | undefined>()
+
+  const [preview, setPreview] = useState<string | undefined>()
+
+  const [progress, setProgress] = useState<number>(0)
+
+  const clearSelectFile = () => {
+    setPreview(undefined)
+  }
+
+  const handleSelectFile = () => {
+    fileInputRef.current?.click()
+  }
+
+  const handleImageLoad = (file: File) => {
+    setImage(file)
+
+    const fileReader = new FileReader()
+
+    fileReader.readAsDataURL(file)
+
+    fileReader.onloadstart = () => {
+      setProgress(0)
+    }
+
+    fileReader.onprogress = (progress) => {
+      setProgress((progress.loaded / progress.total) * 100)
+    }
+
+    fileReader.onload = () => {
+      setPreview(fileReader.result as string)
+      setTimeout(() => setProgress(0), 500)
```

Comment:
Move 500 to constant (optional)

Reply:
ok, I'll name it a.k.a PROGRESS_TIMEOUT_DELAY